apps/backend/src/datasources/mockups/FileDataSource.ts

```diff
@@ -2,52 +2,76 @@ import { ReadStream } from 'pg-large-object';
 
 import { FileMetadata } from '../../models/Blob';
 import { FileDataSource } from '../FileDataSource';
-import { FilesMetadataFilter } from './../../resolvers/queries/FilesMetadataQuery';
 
 export default class FileDataSourceMock implements FileDataSource {
   async prepare(fileId: string, _output: string): Promise<string> {
     return fileId;
   }
 
-  async getMetadata(fileId: string): Promise<FileMetadata>;
-  async getMetadata(filter: FilesMetadataFilter): Promise<FileMetadata[]>;
-  async getMetadata(param: unknown): Promise<FileMetadata | FileMetadata[]> {
-    if (param instanceof FilesMetadataFilter) {
-      const filter = param as FilesMetadataFilter;
-
+  async getMetadata(
+    fileIds?: string[],
+    filenames?: string[],
+    internalUse?: boolean
+  ): Promise<FileMetadata[]> {
+    if (fileIds || filenames) {
```

Comment:
Do we have a dependence on these variables which is outside of this function because the check on filenames is not being used unless l missed it.